app/routes/clients.tsx

```diff
@@ -26,13 +29,39 @@ import { analytics, db } from "~/firebase.config";
 import type { Route } from "./+types/clients";
 import { logEvent } from "firebase/analytics";
 
-export const clientLoader = async () => {
+export const clientLoader = async ({ request }: Route.LoaderArgs) => {
+  // Récupère les paramètres de recherche depuis l'URL
+  const url = new URL(request.url);
+  const searchEmail = url.searchParams.get("email")?.trim() || "";
+  const page = parseInt(url.searchParams.get("page") || "1", 10);
+  const pageSize = 20;
+
   const clientsRef = collection(db, "clients").withConverter(clientConverter);
   const queryRef = query(clientsRef, orderBy("createdAt", "desc"));
   const querySnapshot = await getDocs(queryRef);
-  const clients = querySnapshot.docs.map((doc) => doc.data());
+  let clients = querySnapshot.docs.map((doc) => doc.data());
+
+  // Filtrage par email exact si un email est fourni
+  if (searchEmail) {
+    clients = clients.filter((client) => client.email === searchEmail);
```

Comment:
The search should not be done on client side. See https://firebase.google.com/docs/firestore/query-data/queries for how to filter data from Firebase.